### DISCUSSION TITLE
Support for Hyundia cars in India via India Bluelink

### DISCUSSION THREAD

**Discussion by jaidevpendse** [Hyundai-Kia-Connect/kia_uvo#463] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not able to connect my car in India

**Describe the solution you'd like**
Would love to see the support added for cars in India. I would love to help in all possible ways from my side and would like to a super early tester for the same. 

**Include the logs of the endpoint from Data in HA if this is a sensor**
-NA-

**Describe alternatives you've considered**
-NA-

**Additional context**
-NA-

**Comment by cdnninja** [comment]
This one has come up before.   Someone will need to sniff the traffic and provide so we can reverse engineer this.

**Comment by jaidevpendse** [comment]
> This one has come up before.   Someone will need to sniff the traffic and provide so we can reverse engineer this. 

Should I use something like Wireshark to sniff the packers will it work that way to get the packets from the app traffic

Let me know , If yes will do the research and will provide as much data as possible

**Comment by cdnninja** [comment]
Wireshark won't help since it is encrypted.   This is one method:  https://github.com/fuatakgun/kia_uvo/wiki/How-to-sniff-API-via-MS-Edge

It only works if:

1. A website exists that has all the functionality. 
2. The website uses the same API as the mobile app.  This isn't always the case.

A key for us to be able to make this happen is your implementation not having something called "stamps"  stamps are used in the EU and very difficult to reverse engineer.  Canada and USA doesn't use them but not sure if EU does.

**Comment by cdnninja** [comment]
Here is the canadian API file as an example of what we would replicate for India: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/blob/master/hyundai_kia_connect_api/KiaUvoApiCA.py

**Comment by jaidevpendse** [comment]
> Here is the canadian API file as an example of what we would replicate for India: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/blob/master/hyundai_kia_connect_api/KiaUvoApiCA.py

Thanks for the suggestion and example , will work on it this coming week and will provide as much data as possible

**Comment by cdnninja** [comment]
I will move this over to a discussion for now.  This way it can live on for a longer time if we can't get the details we need.

**Reply by jaidevpendse** [reply]
Unfortunately Bluelink India does not have a web based interface to login , it's just has support for IOS and Android , is there a way to sniff data from these application

**Comment by cdnninja** [comment]
@jaidevpendse  yes https://github.com/fuatakgun/kia_uvo/wiki/How-to-reverse-engineer-app-traffic-on-Android-phone walks through it

**Reply by jaidevpendse** [reply]
Ok will go through it. Thanks
